src/gui/utils.cpp

```diff
@@ -146,7 +146,30 @@ void Utils::Gui::openPath(const Path &path)
     const QUrl url = path.data().startsWith(u"//")
         ? QUrl(u"file:" + path.data())
         : QUrl::fromLocalFile(path.data());
+
+#ifdef Q_OS_WIN
+    auto *thread = QThread::create([path]()
+    {
+        if (SUCCEEDED(::CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE)))
```

Comment:
Why not just use `ShellExecute(nullptr, nullptr, path.toString().toStdWString(), nullptr, nullptr, SW_SHOWNORMAL);`? It is used by `QDesktopServices::openUrl()` when it comes to opening local files/folders.
IIRC, there are no problems with the opening method itself, isn't there? Then the idea is to simply omit all the additional code existing there to support arbitrary URLs, and leave only one that is associated with opening local files/folders. 
IMO, suppressing `SelectItems` in `SHOpenFolderAndSelectItems()` looks like a workaround.

Reply:
Well honesty i used `SHOpenFolderAndSelectItems` again as it is already in the code, so why not. Can't really asses if it is an workaround, if MS made it possible then its probably fine.

Not sure what you mean that there are no problems with the opening method? Opening files? No, its has only issues with folders.

Regardless there was also this related to that code before https://github.com/qbittorrent/qBittorrent/pull/16670

So `ShellExecute` will also need that `thread` and use COM, but yeah i can change it to `ShellExecute`, it will spare some lines, the goal is just for Windows to omit the `QDesktopServices::openUrl`

Reply:
> Not sure what you mean that there are no problems with the opening method?

I mean that most likely there is no problem with using `ShellExecute()` to open folder but delay is introduced by the logic of resolving URL handler that is executed in `QDesktopServices::openUrl()` before `ShellExecute()`.



> Can't really asses if it is an workaround, if MS made it possible then its probably fine.

I didn't find any mention in the documentation that you can use `SHOpenFolderAndSelectItems()` in such a way as to just open a folder without selecting items, so even if it works currently, MS can change it in the future without any notification.

Reply:
Yeah, ofc, the issue is not ShellExecute, but something that happens before, i was unable to pinpoint what it is.
The MS docs are kinda ambiguous for `SHOpenFolderAndSelectItems`, really not sure, they say that cidl param is the count of selection items, but they allow it to be 0, so yeah, anyway, probably better to just use `ShellExecute` and be safe.